test/integration/gardenlet/vpaevictionrequirements/vpaevictionrequirements_test.go

```diff
@@ -0,0 +1,211 @@
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(vpa), vpa)).To(Succeed())
+					g.Expect(vpa.Spec).To(MatchFields(IgnoreExtras, Fields{
+						"UpdatePolicy": PointTo(MatchFields(IgnoreExtras, Fields{
+							"EvictionRequirements": BeEmpty(),
+						})),
+					}))
+				}).Should(Succeed())
+			})
+		})
+
+		When("the Shoot maintenance window is updated", func() {
+			BeforeEach(func() {
+				metav1.SetMetaDataAnnotation(&vpa.ObjectMeta, v1beta1constants.AnnotationShootMaintenanceWindow, maintenanceWindowNow.Begin+","+maintenanceWindowNow.End)
+				vpa.Spec.UpdatePolicy.EvictionRequirements = upscaleOnlyRequirement
+			})
+
+			It("reconciles the VPA again", func() {
+				Eventually(func(g Gomega) {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(vpa), vpa)).To(Succeed())
+					g.Expect(vpa.Spec).To(MatchFields(IgnoreExtras, Fields{
+						"UpdatePolicy": PointTo(MatchFields(IgnoreExtras, Fields{
+							"EvictionRequirements": BeEmpty(),
+						})),
+					}))
+				}).Should(Succeed())
+
+				Expect(testClient.Get(ctx, client.ObjectKeyFromObject(vpa), vpa)).To(Succeed())
+				metav1.SetMetaDataAnnotation(&vpa.ObjectMeta, v1beta1constants.AnnotationShootMaintenanceWindow, maintenanceWindowAlreadyPassed.Begin+","+maintenanceWindowAlreadyPassed.End)
+				Expect(testClient.Update(ctx, vpa)).To(Succeed())
+
+				Eventually(func(g Gomega) {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(vpa), vpa)).To(Succeed())
+					g.Expect(vpa.Spec).To(MatchFields(IgnoreExtras, Fields{
+						"UpdatePolicy": PointTo(MatchFields(IgnoreExtras, Fields{
+							"EvictionRequirements": ConsistOf(upscaleOnlyRequirement),
+						})),
+					}))
+				}).Should(Succeed())
+			})
+		})
+	})
+
+	Context("VPA is labeled with downscale-never", func() {
+		BeforeEach(func() {
+			metav1.SetMetaDataLabel(&vpa.ObjectMeta, v1beta1constants.LabelVPAEvictionRequirementDownscaleRestriction, v1beta1constants.EvictionRequirementNever)
+		})
+
+		It("should add an EvictionRequirement and not requeue, regardless of a Shoot's maintenance window", func() {
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(vpa), vpa)).To(Succeed())
+				g.Expect(vpa.Spec).To(MatchFields(IgnoreExtras, Fields{
+					"UpdatePolicy": PointTo(MatchFields(IgnoreExtras, Fields{
+						"EvictionRequirements": ConsistOf(upscaleOnlyRequirement),
+					})),
+				}))
+			}).Should(Succeed())
+		})
+		When("VPA has an annotation indicating that the Shoot's maintenance window is now", func() {
+			It("doesn't remove the EvictionRequirement", func() {
+				Eventually(func(g Gomega) {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(vpa), vpa)).To(Succeed())
+					g.Expect(vpa.Spec).To(MatchFields(IgnoreExtras, Fields{
+						"UpdatePolicy": PointTo(MatchFields(IgnoreExtras, Fields{
+							"EvictionRequirements": ConsistOf(upscaleOnlyRequirement),
+						})),
+					}))
+				}).Should(Succeed())
+			})
```

Comment:
This test is exactly the same as the one in L188-L197. We don't see any different setup in a BeforeEach. Maybe you wanted to have a BeforeEach to have a different setup. Otherwise both test run with the same test content and with the same setup. Anything we miss?

Reply:
Great find! I was missing additional setup in an `BeforeEach` block here!